src/Evolution/Systems/RadiationTransport/M1Grey/BoundaryCorrections/BoundaryCorrection.hpp

```diff
@@ -0,0 +1,44 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <memory>
+#include <pup.h>
+
+#include "Parallel/CharmPupable.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// Boundary corrections/numerical fluxes
+namespace RadiationTransport::M1Grey::BoundaryCorrections {
+/// \cond
+template <typename... NeutrinoSpecies>
+class Rusanov;
+/// \endcond
+
+/*!
+ * \brief The base class used to create boundary corrections from input files
+ * and store them in the global cache.
+ */
+template <typename... NeutrinoSpecies>
+class BoundaryCorrection : public PUP::able {
```

Comment:
(not for this PR) Like for boundary conditions I think this base class has to refer to documentation on the compile-time interface that its derived classes implement. Else it's impossible to understand what the various required type aliases mean, how they connect to the function signatures, and what is expected of the functions. I think the possibilities to think about are: (1) A documentation-link only, to some place in the docs that explain the interface, (2) a base class for all evolution systems that documents the interface, (3) a protocol that documents the interface and maybe also does a few consistency checks.